keyboards/chromatonemini/keymaps/default/keymap.c

```diff
@@ -0,0 +1,122 @@
+            MI_BENDU,                                                                                                                                                                            \
+        SHIFTDN, SHIFTUP,    MI_C_1, MI_D_1, MI_E_1,  MI_Fs_1, MI_Ab_1, MI_Bb_1, MI_C_2, MI_D_2, MI_E_2, MI_Fs_2, MI_Ab_2, MI_Bb_2, MI_C_3,  MI_D_3,  MI_E_3, MI_Fs_3, MI_Ab_3, MI_Bb_3, MI_C_4, \
+            MI_BENDD,           MI_Db_1, MI_Eb_1, MI_F_1,  MI_G_1,  MI_A_1,  MI_B_1, MI_Db_2, MI_Eb_2, MI_F_2,  MI_G_2, MI_A_2,  MI_B_2, MI_Db_3, MI_Eb_3, MI_F_3,  MI_G_3,  MI_A_3,  MI_B_3     \
+    ),
+
+    /* Trans   This layer must locate 1 layer above _BASE layer. */
+    [_TRANS] = LAYOUT( \
+            _______,                                                                                                                                                                                     \
+            _______,                                                                                                                                                                                     \
+        MI_TRNSD, MI_TRNSU,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+            _______,               _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______      \
+    ),
+
+    [_FN] =  LAYOUT( \
+            _______,                                                                                                                                                                                     \
+            MI_VELU,                                                                                                                                                                                     \
+        MI_OCTD, MI_OCTU,     L_BASE, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,  \
+            MI_VELD,               L_TRANS, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, RGB_SAD, RGB_HUD, RGB_SPD, RGB_VAD, RGB_RMOD, TGLINDI     \
+    )
+};
+
+void keyboard_post_init_user(void) {
+    //  Set otave to MI_OCT_0
+    midi_config.octave = MI_OCT_0 - MIDI_OCTAVE_MIN;
+
+    default_layer_set(1UL << _BASE);
+};
+
+void reset_scale_indicator(void) {
+    //  reset transpose value and scale_indicator_col to default.
+    midi_config.transpose = 0;
+    scale_indicator_col = DEFAULT_SCALE_COL;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+
+        case L_BASE:
+            reset_scale_indicator();
+            default_layer_set(1UL << _BASE);
```

Comment:
Just a heads up, you can use `DF(_BASE)` in the keymap to accomplish this.  And then you could use `default_layer_state_set_user` to handle the additional stuff.

Reply:
Thank you for the heads up. I didn't know that. Well, particularly for here, it might be better for an amateur like me to keep it as it is to show what would happen in one area.

Reply:
That's fair.